api/db/migrations/20201113105430_migrate-password-and-shouldChangePassword-to-authentication-methods.js

```diff
@@ -0,0 +1,48 @@
+const { batch } = require('../batchTreatment');
+
+const AuthenticationMethod = require('../../lib/domain/models/AuthenticationMethod');
+
+const TABLE_USERS = 'users';
+const PASSWORD_COLUMN = 'password';
+const SHOULD_CHANGE_PASSWORD_COLUMN = 'shouldChangePassword';
+
+const TABLE_AUTHENTICATION_METHODS = 'authentication-methods';
+
+exports.up = async (knex) => {
+  const sqlRequest = `SELECT "id", "${PASSWORD_COLUMN}", "${SHOULD_CHANGE_PASSWORD_COLUMN}" ` +
+                     `FROM "${TABLE_USERS}" ` +
+                     `WHERE ("${PASSWORD_COLUMN}" = '') IS FALSE`;
+
+  const usersWithPasswordInformations = await knex.raw(sqlRequest);
+
+  return batch(knex, usersWithPasswordInformations.rows, (user) => {
```

Comment:
Problème: l'instruction batch ici ne gère que des lots de 10 éléments ce qui va rendre le script beaucoup trop long à exécuter. Le plus simple serait d'utiliser une requête INSERT INTO SELECT qui va se charger d'effectuer toutes les insertions en une seule fois:
`INSERT INTO "authentication-methods"("identityProvider", "userId", "authenticationComplement") SELECT 'PIX', id, jsonb_build_object('password', "password", 'shouldChangePassword', "shouldChangePassword") FROM "users" WHERE ("password" = '') IS FALSE;`

Reply:
Done